src/vm/script.rs

```diff
@@ -1129,12 +1156,45 @@ impl<'a> ScriptExecutor<'a> {
                         exec_stack.swap_unchecked(len - 2, len - 6);
 
                         // We then swap the third and fourth elements with the last two:
-                        // [4, 5, 0, 1, 2, 3]
+                        // [2, 3, 4, 5, 0, 1]
```

Comment:
This was right but I was seeing the top of the stack as the leftmost element. However, this is more in line with the vm implementation.